src/Transport/Administration/QueueCreator.cs

```diff
@@ -0,0 +1,115 @@
+
+            var topic = new TopicDescription(topicName)
+            {
+                EnableBatchedOperations = true,
+                EnablePartitioning = enablePartitioning,
+                MaxSizeInMB = maxSizeInMB
+            };
+
+            try
+            {
+                await client.CreateTopicAsync(topic).ConfigureAwait(false);
+            }
+            catch (MessagingEntityAlreadyExistsException)
+            {
+            }
+            // TODO: refactor when https://github.com/Azure/azure-service-bus-dotnet/issues/525 is fixed
+            catch (ServiceBusException sbe) when (sbe.Message.Contains("SubCode=40901.")) // An operation is in progress.
+            {
+            }
+
+            foreach (var address in queueBindings.ReceivingAddresses.Concat(queueBindings.SendingAddresses))
+            {
+                var queue = new QueueDescription(address)
+                {
+                    EnableBatchedOperations = true,
+                    LockDuration = TimeSpan.FromMinutes(5),
+                    MaxDeliveryCount = int.MaxValue,
+                    MaxSizeInMB = maxSizeInMB,
+                    EnablePartitioning = enablePartitioning
+                };
+
+                try
+                {
+                    await client.CreateQueueAsync(queue).ConfigureAwait(false);
+                }
+                catch (MessagingEntityAlreadyExistsException)
+                {
+                }
+                // TODO: refactor when https://github.com/Azure/azure-service-bus-dotnet/issues/525 is fixed
+                catch (ServiceBusException sbe) when (sbe.Message.Contains("SubCode=40901.")) // An operation is in progress.
+                {
+                }
+            }
+
+            var subscriptionName = mainInputQueueName.Length > maxNameLength ? subscriptionShortener(mainInputQueueName) : mainInputQueueName;
+            var subscription = new SubscriptionDescription(topicName, subscriptionName)
+            {
+                LockDuration = TimeSpan.FromMinutes(5),
+                ForwardTo = mainInputQueueName,
+                EnableDeadLetteringOnFilterEvaluationExceptions = false,
+                MaxDeliveryCount = int.MaxValue,
+                EnableBatchedOperations = true,
+                UserMetadata = mainInputQueueName
+            };
+
+            try
+            {
+                await client.CreateSubscriptionAsync(subscription, new RuleDescription("$default", new FalseFilter())).ConfigureAwait(false);
+            }
+            catch (MessagingEntityAlreadyExistsException)
+            {
+            }
+            // TODO: refactor when https://github.com/Azure/azure-service-bus-dotnet/issues/525 is fixed
+            catch (ServiceBusException sbe) when (sbe.Message.Contains("SubCode=40901.")) // An operation is in progress.
+            {
+            }
+
+            await client.CloseAsync().ConfigureAwait(false);
```

Comment:
Should this be in a finally block?

Reply:
If not mistaken, we wanted to throw if would encounter one of the exceptions that are not "whitelisted". And if we'd throw, would there be still a point in closing the client in finally?